packages/cadl-ranch-specs/http/lro/lro-core/main.cadl

```diff
@@ -0,0 +1,204 @@
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+import "@cadl-lang/rest";
+
+using Azure.Core;
+using global.Azure.Core.Traits;
+using Cadl.Rest;
+using Cadl.Http;
+
+#suppress "@azure-tools/cadl-azure-core/casing-style" "For spec"
+@doc("Illustrates bodies templated with Azure Core with long-running operation")
+@scenarioService("/azure/lro/core")
+namespace Azure.Lro.Core;
+
+interface ResourceOperations
+  extends global.Azure.Core.ResourceOperations<NoConditionalRequests & NoRepeatableRequests & NoClientRequestId> {}
+
+@resource("users")
+@doc("Details about a user.")
+model User {
+  @key
+  @visibility("read")
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The role of user")
+  role: string;
+}
+
+@doc("The parameters for exporting a user.")
+model UserExportParams {
+  @query
+  @doc("The format of the data.")
+  format: string;
+}
+
+@doc("The exported user data.")
+model ExportedUser {
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The exported URI.")
+  resourceUri: string;
+}
+
+@scenario
+@doc("Creates or repalces a User")
+@summary("Adds a user or repalces a user's fields.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected verb: PUT
+Expected path parameter: name=madge
+
+Expected request body:
+```json
+{
+  "role": "contributor"
+}
+```
+
+Expected status code: 201
+Expected response header: operation-location={endpoint}/users/madge/operations/operation1
```

Comment:
The status monitor URL from current Azure.Core is not exactly same as from the Guideline (note the format of the status monitor URL is not a requirement from Guideline).

This URL should be opaque to SDK, so for now I adopt the one from `Azure.Core.ResourceOperations.GetResourceOperationStatus<User, ..>`